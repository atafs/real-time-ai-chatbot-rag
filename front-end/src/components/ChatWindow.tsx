// src/components/ChatWindow.tsx
import React from "react";
import { Message } from "../types";

interface ChatWindowProps {
  messages: Message[];
}

const ChatWindow: React.FC<ChatWindowProps> = ({ messages }) => {
  return (
    <div
      className="chat border border-gray-300 p-4 h-96 overflow-y-scroll mb-4"
      role="log"
      aria-live="polite"
    >
      {messages.map((msg, i) => (
        <p
          key={i}
          className={`mb-2 ${
            msg.sender === "user"
              ? "text-right text-blue-600"
              : "text-left text-green-600"
          }`}
        >
          <span className="font-bold">{msg.sender}: </span>
          {msg.text}{" "}
          <span className="text-xs text-gray-500">({msg.timestamp})</span>
        </p>
      ))}
    </div>
  );
};

export default ChatWindow;
