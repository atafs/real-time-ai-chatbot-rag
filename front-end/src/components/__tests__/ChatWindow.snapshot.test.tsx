import { render } from "@testing-library/react";
import ChatWindow from "../ChatWindow";

test("ChatWindow snapshot", () => {
  const messages = [
    { sender: "user", text: "Hello!", timestamp: "10:00" },
    { sender: "bot", text: "Hi!", timestamp: "10:01" },
  ];
  const { asFragment } = render(<ChatWindow messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});
