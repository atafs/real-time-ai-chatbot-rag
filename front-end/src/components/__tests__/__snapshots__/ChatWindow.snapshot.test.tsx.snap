// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindow snapshot 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="chat border border-gray-300 p-4 h-96 overflow-y-scroll mb-4"
    role="log"
  >
    <p
      class="mb-2 text-right text-blue-600"
    >
      <span
        class="font-bold"
      >
        user: 
      </span>
      Hello! 
      <span
        class="text-xs text-gray-500"
      >
        (10:00)
      </span>
    </p>
    <p
      class="mb-2 text-left text-green-600"
    >
      <span
        class="font-bold"
      >
        bot: 
      </span>
      Hi! 
      <span
        class="text-xs text-gray-500"
      >
        (10:01)
      </span>
    </p>
  </div>
</DocumentFragment>
`;
